### PR TITLE
protobuf 3.17.0

### PR DESCRIPTION
**Pull request by Loyalsoldier**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by carlocab** [comment]
Will probably have the same failure for `mavsdk` as #76858. Solutions are:

1. Wait for a `mavsdk` version bump
2. Let `mavsdk` vendor all its dependencies (Sorry, @Bo98. No option to selectively vendor them, I think.)
3. Find a way to regenerate the relevant `mavsdk` source files. See https://github.com/Homebrew/homebrew-core/pull/76858#issuecomment-839159776.

**Comment by Loyalsoldier** [comment]
mavsdk has a new release https://github.com/Homebrew/homebrew-core/pull/77375

**Comment by Loyalsoldier** [comment]
@carlocab Please cancel these workflows and wait for the new mavsdk.

**Comment by carlocab** [comment]
Might be better to include the `mavsdk` version bump here. That way we know right away if the new version works with this version of `protobuf`.

**Comment by carlocab** [comment]
I've triggered a merge for #77375. You can rebase this shortly.

**Comment by cho-m** [comment]
New `mavsdk` might not help.

They vendored protobuf v3.15.3: https://github.com/mavlink/MAVSDK/blob/v0.40.0/third_party/protobuf/CMakeLists.txt#L39

Here is the current protobuf version check: https://github.com/mavlink/MAVSDK/blob/v0.40.0/src/mavsdk_server/src/generated/mavsdk_options.pb.h#L16
```h
#if 3015003 < PROTOBUF_MIN_PROTOC_VERSION
#error This file was generated by an older version of protoc which is
#error incompatible with your Protocol Buffer headers. Please
```

**Comment by carlocab** [comment]
> New `mavsdk` might not help.

I took a quick look at it earlier and came to the same conclusion. Anyone know how to regenerate these headers?

**Comment by carlocab** [comment]
Seems like they just run the script without any additional setup:

https://github.com/mavlink/MAVSDK/blob/988a10244856324a622007546d3231b5dfe1d475/.github/workflows/main.yml#L118-L119

They to build `mavsdk` first. It would be unfortunate if you need a working `mavsdk` before you can regenerate those headers.

**Comment by cho-m** [comment]
The script includes the vendored path to `protoc`:
https://github.com/mavlink/MAVSDK/blob/v0.40.0/tools/generate_from_protos.sh#L10

Also, an annoying part will be the custom plugin they use: https://pypi.org/project/protoc-gen-mavsdk/
If this is required for regeneration, this means adding in a Python dependency and the resource.

I also found some instructions on running script for new plugins.
I think it is used for the existing/default plugins in repo:
https://mavsdk.mavlink.io/main/en/cpp/contributing/plugins.html#generate-h-and-cpp-files

**Comment by carlocab** [comment]
Even worse, they just provide a wheel for `protoc-gen-mavsdk`, so we can't even `venv.pip_install` it.

**Comment by carlocab** [comment]
Tried this, doesn't work (but I probably just don't understand `pip`):

<details><summary>Diff</summary>
<p>

```diff
diff --git a/Formula/mavsdk.rb b/Formula/mavsdk.rb
index 4483bdd951..3d4a20f979 100644
--- a/Formula/mavsdk.rb
+++ b/Formula/mavsdk.rb
@@ -1,4 +1,6 @@
 class Mavsdk < Formula
+  include Language::Python::Virtualenv
+
   desc "API and library for MAVLink compatible systems written in C++17"
   homepage "https://mavsdk.mavlink.io"
   url "https://github.com/mavlink/MAVSDK.git",
@@ -18,6 +20,8 @@ class Mavsdk < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "python@3.9" => :build
+  depends_on "six" => :build
   depends_on "abseil"
   depends_on "c-ares"
   depends_on "curl"
@@ -31,7 +35,43 @@ class Mavsdk < Formula
 
   uses_from_macos "zlib"
 
+  resource "Jinja2" do
+    url "https://files.pythonhosted.org/packages/7a/0c/23cbcf515b5394e9f59a3e6629f26e1142b92d474ee0725a26aa5a3bcf76/Jinja2-3.0.0.tar.gz"
+    sha256 "ea8d7dd814ce9df6de6a761ec7f1cac98afe305b8cdc4aaae4e114b8d8ce24c5"
+  end
+
+  resource "MarkupSafe" do
+    url "https://files.pythonhosted.org/packages/67/6a/5b3ed5c122e20c33d2562df06faf895a6b91b0a6b96a4626440ffe1d5c8e/MarkupSafe-2.0.0.tar.gz"
+    sha256 "4fae0677f712ee090721d8b17f412f1cbceefbf0dc180fe91bab3232f38b4527"
+  end
+
+  resource "protoc-gen-mavsdk" do
+    url "https://files.pythonhosted.org/packages/da/91/96d4bde10cf9512250290880373fec0e1e068a9cafae96e69e90f45b2616/protoc_gen_mavsdk-1.0.1-py3-none-any.whl"
+    sha256 "5ccdfadea69e2da95567a8aa1925bdb2d34697def2ce1d5835b596c97abcc1ac"
+  end
+
   def install
+    # We need `protoc-gen-mavsdk` to bootstrap
+    venv = virtualenv_create(buildpath/"bootstrap", "python3")
+    venv.pip_install %w[Jinja2 MarkupSafe]
+    buildpath.install resource("protoc-gen-mavsdk")
+    system buildpath/"bootstrap/bin/python", "-m", "pip", "install", "-v",
+           "--no-deps",
+           "--no-index",
+           "--upgrade",
+           "--isolated",
+           "--target=#{buildpath}/pgm",
+           Dir[buildpath/"protoc_gen_mavsdk-*.whl"].first
+    ENV.prepend_path "PATH", buildpath/"pgm/bin"
+    ENV.prepend_path "PATH", buildpath/"bootstrap/bin"
+
+    generator = "tools/generate_from_protos.sh"
+    inreplace generator do |s|
+      s.gsub!(/^(protoc_binary)=.*/, "\\1=#{Formula["protobuf"].opt_bin}/protoc")
+      s.gsub!(/^(protoc_grpc_binary)=.*/, "\\1=#{Formula["grpc"].opt_bin}/grpc_cpp_plugin")
+    end
+    system generator
+
     # Source build adapted from
     # https://mavsdk.mavlink.io/develop/en/contributing/build.html
     system "cmake", *std_cmake_args,
```

</p>
</details>

Fails with the error
```
==> tools/generate_from_protos.sh
Found protoc (libprotoc 3.15.8): /usr/local/opt/protobuf/bin/protoc
Found grpc_cpp_plugin: /usr/local/opt/grpc/bin/grpc_cpp_plugin
Processing mavsdk_options.proto
Processing action/action.proto
Traceback (most recent call last):
  File "/private/tmp/mavsdk-20210516-89791-163osqy/pgm/bin/protoc-gen-mavsdk", line 5, in <module>
    from protoc_gen_mavsdk.__main__ import main
ModuleNotFoundError: No module named 'protoc_gen_mavsdk'
--custom_out: protoc-gen-custom: Plugin failed with status code 1.
```

**Comment by carlocab** [comment]
#77398

**Comment by cho-m** [comment]
@carlocab
For other way, I think you need to modify pythonpath, maybe something like:
`ENV.prepend_path "PYTHONPATH", buildpath/"pgm"`

Looks like ARM tests finished:
- As expected, `mavsdk` failed
- `mosh` seems to be hitting `<version>` header issue

**Comment by carlocab** [comment]
Tried this, also doesn't work:

<details><summary>Diff</summary>
<p>

```diff
diff --git a/Formula/mavsdk.rb b/Formula/mavsdk.rb
index 4483bdd951..8a88bd6f66 100644
--- a/Formula/mavsdk.rb
+++ b/Formula/mavsdk.rb
@@ -1,4 +1,6 @@
 class Mavsdk < Formula
+  include Language::Python::Virtualenv
+
   desc "API and library for MAVLink compatible systems written in C++17"
   homepage "https://mavsdk.mavlink.io"
   url "https://github.com/mavlink/MAVSDK.git",
@@ -18,6 +20,8 @@ class Mavsdk < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "python@3.9" => :build
+  depends_on "six" => :build
   depends_on "abseil"
   depends_on "c-ares"
   depends_on "curl"
@@ -31,7 +35,48 @@ class Mavsdk < Formula
 
   uses_from_macos "zlib"
 
+  resource "Jinja2" do
+    url "https://files.pythonhosted.org/packages/7a/0c/23cbcf515b5394e9f59a3e6629f26e1142b92d474ee0725a26aa5a3bcf76/Jinja2-3.0.0.tar.gz"
+    sha256 "ea8d7dd814ce9df6de6a761ec7f1cac98afe305b8cdc4aaae4e114b8d8ce24c5"
+  end
+
+  resource "MarkupSafe" do
+    url "https://files.pythonhosted.org/packages/67/6a/5b3ed5c122e20c33d2562df06faf895a6b91b0a6b96a4626440ffe1d5c8e/MarkupSafe-2.0.0.tar.gz"
+    sha256 "4fae0677f712ee090721d8b17f412f1cbceefbf0dc180fe91bab3232f38b4527"
+  end
+
+  resource "protoc-gen-mavsdk" do
+    url "https://files.pythonhosted.org/packages/da/91/96d4bde10cf9512250290880373fec0e1e068a9cafae96e69e90f45b2616/protoc_gen_mavsdk-1.0.1-py3-none-any.whl"
+    sha256 "5ccdfadea69e2da95567a8aa1925bdb2d34697def2ce1d5835b596c97abcc1ac"
+  end
+
   def install
+    generator = "tools/generate_from_protos.sh"
+    inreplace generator do |s|
+      s.gsub!(/^(protoc_binary)=.*/, "\\1=#{Formula["protobuf"].opt_bin}/protoc")
+      s.gsub!(/^(protoc_grpc_binary)=.*/, "\\1=#{Formula["grpc"].opt_bin}/grpc_cpp_plugin")
+    end
+
+    # We need `protoc-gen-mavsdk` to bootstrap
+    venv_dir = buildpath/"bootstrap"
+    with_env(
+      VIRTUAL_ENV: venv_dir,
+      PYTHONPATH:  venv_dir,
+      PATH:        "#{venv_dir}/bin:#{ENV["PATH"]}",
+    ) do
+      buildpath.install resource("protoc-gen-mavsdk")
+      system "python3", "-m", "pip", "install", "-v",
+             "--no-deps",
+             "--no-index",
+             "--upgrade",
+             "--isolated",
+             "--target=#{venv_dir}",
+             Dir[buildpath/"protoc_gen_mavsdk-*.whl"].first
+      venv = virtualenv_create(venv_dir, "python3")
+      venv.pip_install %w[Jinja2 MarkupSafe]
+      system generator
+    end
+
     # Source build adapted from
     # https://mavsdk.mavlink.io/develop/en/contributing/build.html
     system "cmake", *std_cmake_args,
```

</p>
</details>

Fails with the error
```
==> tools/generate_from_protos.sh
Found protoc (libprotoc 3.15.8): /usr/local/opt/protobuf/bin/protoc
Found grpc_cpp_plugin: /usr/local/opt/grpc/bin/grpc_cpp_plugin
Processing mavsdk_options.proto
Processing action/action.proto
Traceback (most recent call last):
  File "/private/tmp/mavsdk-20210516-11045-2s47ij/bootstrap/bin/protoc-gen-mavsdk", line 5, in <module>
    from protoc_gen_mavsdk.__main__ import main
  File "/private/tmp/mavsdk-20210516-11045-2s47ij/bootstrap/protoc_gen_mavsdk/__init__.py", line 5, in <module>
    from .autogen import AutoGen
  File "/private/tmp/mavsdk-20210516-11045-2s47ij/bootstrap/protoc_gen_mavsdk/autogen.py", line 4, in <module>
    from .autogen_file import File
  File "/private/tmp/mavsdk-20210516-11045-2s47ij/bootstrap/protoc_gen_mavsdk/autogen_file.py", line 2, in <module>
    from .utils import name_parser_factory
  File "/private/tmp/mavsdk-20210516-11045-2s47ij/bootstrap/protoc_gen_mavsdk/utils.py", line 2, in <module>
    from jinja2 import Environment, FileSystemLoader
ModuleNotFoundError: No module named 'jinja2'
--custom_out: protoc-gen-custom: Plugin failed with status code 1.
```

**Comment by carlocab** [comment]
Ah, wait. I think I have it.

**Comment by carlocab** [comment]
#77399

Thanks for the tip, @cho-m.

**Comment by cho-m** [comment]
In the other failure, `mosh` generates the `VERSION` file during `make`.
```makefile
VERSION:
	@echo mosh 1.3.2 > VERSION.dist
	@set -e; if git describe --dirty --always > VERSION.git 2>&1 && \
		[ -z `git rev-parse --show-prefix` ]; then \
		if ! diff -q VERSION.git VERSION > /dev/null 2>&1; then \
			mv -f VERSION.git VERSION; \
		fi; \
	elif ! diff -q VERSION.dist VERSION > /dev/null 2>&1; then \
		mv -f VERSION.dist VERSION; \
	fi
	@rm -f VERSION.dist VERSION.git
```

May need to create patch to rename file.
Not too sure if there is another alternative.

**Comment by carlocab** [comment]
Cancelling the tests since this is blocked by #77399.

File: Formula/wownero.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class Wownero < Formula
       tag:      "v0.9.3.3",
       revision: "e2d2b9a447502e22467af9df20e0732b3dd4ac4c"
   license "BSD-3-Clause"
+  revision 1
```
Comment (carlocab) [inline Formula/wownero.rb:8]: This reports no linkage with `protobuf`, so it's unlikely that a revision bump is needed.

File: Formula/dnsdist.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@ class Dnsdist < Formula
   homepage "https://www.dnsdist.org/"
   url "https://downloads.powerdns.com/releases/dnsdist-1.6.0.tar.bz2"
   sha256 "a7783a04d8d4ad2b0168ffaaf85ef95d5f557057b0462280684dd799d0cdd292"
+  revision 1
```
Comment (carlocab) [inline Formula/dnsdist.rb:6]: This has no linkage with `protobuf`, so a revision bump is probably unneeded.

File: Formula/dvc.rb [hunk 3]
```diff
@@ -6,6 +6,7 @@ class Dvc < Formula
   url "https://files.pythonhosted.org/packages/0f/3b/d51f69c500718eb65875dc8134370d1aedf1983327e284ab1464fc36c85a/dvc-2.1.0.tar.gz"
   sha256 "46cfbf0db27107fb3a2d5c643e3a948bb24539bf165ef70e77ce64283959e481"
   license "Apache-2.0"
+  revision 1
```
Comment (carlocab) [inline Formula/dvc.rb:9]: This does not report linkage with protobuf.

**Review by carlocab — DISMISSED** [review]
Let's keep revision bumps to a minimum please. Users complain often about unexpected formula upgrades.

**Comment by carlocab** [comment]
I've merged #77399, so I rebased this for you, cleaned up your commits, and dropped the ones that I think won't be needed. (We'll find out soon if I was wrong.)

**Comment by cho-m** [comment]
@carlocab Is a fix needed for `mosh` version header failure? The Big Sur nodes are probably going to fail on it.

**Comment by carlocab** [comment]
> Is a fix needed for `mosh` version header failure? The Big Sur nodes are probably going to fail on it.

`mosh` has linkage with `protobuf`, so yes, we'll need one.

**Comment by carlocab** [comment]
Have you worked out when `mosh` uses the `VERSION` file?

**Comment by cho-m** [comment]
The upstream doesn't seem very active (https://github.com/mobile-shell/mosh).
Current `mosh` release is from 2017. Last activity was May 2020.

I'll also see if we can try patching the VERSION usage to VERSION.txt or something.

Sadly due to VERSION file being created by Makefile, there I don't know of an easy fix like changing build directory.

**Comment by carlocab** [comment]
I think they're just taking their time for a new release: https://github.com/mobile-shell/mosh/issues/1132

We might at least get a suggestion for all the places that need fixing from a bug report.

**Comment by cho-m** [comment]
My attempt at patching the generated `Makefile.in` files from the tarball (had git init the tarball files for diff).
(the actual code change upstream may need to be done on the `Makefile.am` files):
```patch
diff --git a/Makefile.in b/Makefile.in
index 2de6cc8..f77ae3b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -811,23 +811,23 @@ VERSION:
 	@echo @PACKAGE_STRING@ > VERSION.dist
 	@set -e; if git describe --dirty --always > VERSION.git 2>&1 && \
 		[ -z `git rev-parse --show-prefix` ]; then \
-		if ! diff -q VERSION.git VERSION > /dev/null 2>&1; then \
-			mv -f VERSION.git VERSION; \
+		if ! diff -q VERSION.git VERSION.txt > /dev/null 2>&1; then \
+			mv -f VERSION.git VERSION.txt; \
 		fi; \
-	elif ! diff -q VERSION.dist VERSION > /dev/null 2>&1; then \
-		mv -f VERSION.dist VERSION; \
+	elif ! diff -q VERSION.dist VERSION.txt > /dev/null 2>&1; then \
+		mv -f VERSION.dist VERSION.txt; \
 	fi
 	@rm -f VERSION.dist VERSION.git
 
 version.h:	VERSION
-	@printf '#define BUILD_VERSION "%s"\n' "$$(cat VERSION)" > version.h.new
+	@printf '#define BUILD_VERSION "%s"\n' "$$(cat VERSION.txt)" > version.h.new
 	@set -e; if ! diff -q version.h version.h.new > /dev/null 2>&1; then \
 		mv -f version.h.new version.h; \
 	fi
 	@rm -f version.h.new
 
 clean-local:
-	@rm -rf version.h VERSION cov-int mosh-coverity.txz
+	@rm -rf version.h VERSION.txt cov-int mosh-coverity.txz
 
 cppcheck: $(BUILT_SOURCES) config.h
 	cppcheck --enable=all --template=gcc -include config.h -I . \
diff --git a/scripts/Makefile.in b/scripts/Makefile.in
index 13d261e..45261e5 100644
--- a/scripts/Makefile.in
+++ b/scripts/Makefile.in
@@ -513,8 +513,8 @@ uninstall-am: uninstall-binSCRIPTS
 .PRECIOUS: Makefile
 
 
-mosh:	mosh.pl ../VERSION Makefile
-	@sed -e "s/\@VERSION\@/`cat ../VERSION`/" -e "s/\@PACKAGE_STRING\@/@PACKAGE_STRING@/" $(srcdir)/mosh.pl > mosh
+mosh:	mosh.pl ../VERSION.txt Makefile
+	@sed -e "s/\@VERSION\@/`cat ../VERSION.txt`/" -e "s/\@PACKAGE_STRING\@/@PACKAGE_STRING@/" $(srcdir)/mosh.pl > mosh
 	@chmod a+x mosh
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
```

**Comment by carlocab** [comment]
Does it build, and generate the correct version information? If so, we can probably use it, but we'll want to report the error upstream first.

**Comment by cho-m** [comment]
I did build and tried basic commands, but haven't actually tried connecting between `mosh` client and server:
```console
❯ brew install mosh --build-from-source
...
...
==> Patching
==> Applying e5f8a826ef9ff5da4cfce3bb8151f9526ec19db0.patch
patching file src/network/network.cc
Hunk #1 succeeded at 335 with fuzz 1 (offset 3 lines).
==> ./configure --prefix=/opt/homebrew/Cellar/mosh/1.3.2_15 --enable-completion
==> make install
🍺  /opt/homebrew/Cellar/mosh/1.3.2_15: 15 files, 912.8KB, built in 17 seconds

❯ brew test mosh
==> Testing mosh
==> /opt/homebrew/Cellar/mosh/1.3.2_15/bin/mosh-client -c

❯ echo $?
0

❯ mosh --version
mosh 1.3.2 [build mosh 1.3.2]
Copyright 2012 Keith Winstein <mosh-devel@mit.edu>
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by cho-m** [comment]
Can check if I included all necessary details on upstream issue mobile-shell/mosh#1140

**Comment by carlocab** [comment]
Nope, looks good. Thanks @cho-m.

**Comment by cho-m** [comment]
@carlocab Got a quick response with link to a commit in master branch: https://github.com/mobile-shell/mosh/commit/12199114fe4234f791ef4c306163901643b40538

We can try to patch with it. We will need to add in extra autotool dependencies to rebuild files.

**Comment by carlocab** [comment]
> We can try to patch with it. We will need to add in extra autotool dependencies to rebuild files.

We can do that. Does the patch from https://github.com/mobile-shell/mosh/commit/12199114fe4234f791ef4c306163901643b40538 apply cleanly?

**Comment by carlocab** [comment]
Turns out this bug has been known for a while: https://bugs.llvm.org/show_bug.cgi?id=42540

**Comment by cho-m** [comment]
> Does the patch from [mobile-shell/mosh@1219911](https://github.com/mobile-shell/mosh/commit/12199114fe4234f791ef4c306163901643b40538) apply cleanly?

Didn't work for me.
I guess the 3 year gap between the **1.3.2** release and this commit are causing issues.

We probably still need to create a new patch.

**Comment by cho-m** [comment]
`mavsdk` still failed, but I think the error is farther along now.
Maybe it isn't regenerating some files and need to delete the existing files?
May need to try locally.
```make
/tmp/mavsdk-20210516-33825-kxcka2/src/mavsdk_server/src/generated/telemetry/telemetry.pb.h:17:2: error: This file was generated by an older version of protoc which is
#error This file was generated by an older version of protoc which is
 ^
```

---

For `mosh`, will probably need to build a local patch either way.
`Makefile.am` and `configure.ac` have changed since stable release.
Options I got working:
<details>
<summary>mosh.rb diff 1 based on upstream fix</summary>

```patch
diff --git a/Formula/mosh.rb b/Formula/mosh.rb
index 515ab1925a..25f2c82072 100644
--- a/Formula/mosh.rb
+++ b/Formula/mosh.rb
@@ -3,6 +3,7 @@ class Mosh < Formula
   homepage "https://mosh.org"
   license "GPL-3.0-or-later"
   revision 15
+  head "https://github.com/mobile-shell/mosh.git", shallow: false

   stable do
     url "https://mosh.org/mosh-1.3.2.tar.gz"
@@ -13,6 +14,9 @@ class Mosh < Formula
       url "https://github.com/mobile-shell/mosh/commit/e5f8a826ef9ff5da4cfce3bb8151f9526ec19db0.patch?full_index=1"
       sha256 "022bf82de1179b2ceb7dc6ae7b922961dfacd52fbccc30472c527cb7c87c96f0"
     end
+
+    # Fix big sur build.
+    patch :DATA
   end

   bottle do
@@ -23,13 +27,8 @@ class Mosh < Formula
     sha256 cellar: :any, mojave:        "f921eb384a179c6b65078a36bc479087c20de42562957d92510dc54dfec92c97"
   end

-  head do
-    url "https://github.com/mobile-shell/mosh.git", shallow: false
-
-    depends_on "autoconf" => :build
-    depends_on "automake" => :build
-  end
-
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
   depends_on "pkg-config" => :build
   depends_on "openssl@1.1"
   depends_on "protobuf"
@@ -45,7 +44,7 @@ class Mosh < Formula
     # PATH to support launching outside shell e.g. via launcher
     inreplace "scripts/mosh.pl", "'mosh-client", "\'#{bin}/mosh-client"

-    system "./autogen.sh" if build.head?
+    system "./autogen.sh"
     system "./configure", "--prefix=#{prefix}", "--enable-completion"
     system "make", "install"
   end
@@ -54,3 +53,87 @@ class Mosh < Formula
     system bin/"mosh-client", "-c"
   end
 end
+
+__END__
+diff --git a/Makefile.am b/Makefile.am
+index ac0bb8f..52726c6 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -1,7 +1,7 @@
+ ACLOCAL_AMFLAGS = -I m4
+ SUBDIRS = scripts src man conf
+ EXTRA_DIST = autogen.sh ocb-license.html README.md COPYING.iOS
+-BUILT_SOURCES = version.h
++BUILT_SOURCES = VERSION
+ CLANG_SCAN_BUILD = scan-build
+ AM_DISTCHECK_CONFIGURE_FLAGS = --enable-compile-warnings=distcheck --enable-examples
+
+@@ -19,18 +19,11 @@ VERSION:
+ 	fi
+ 	@rm -f VERSION.dist VERSION.git
+
+-version.h:	VERSION
+-	@printf '#define BUILD_VERSION "%s"\n' "$$(cat VERSION)" > version.h.new
+-	@set -e; if ! diff -q version.h version.h.new > /dev/null 2>&1; then \
+-		mv -f version.h.new version.h; \
+-	fi
+-	@rm -f version.h.new
+-
+ clean-local:
+-	@rm -rf version.h VERSION cov-int mosh-coverity.txz
++	@rm -rf VERSION cov-int mosh-coverity.txz
+
+ cppcheck: $(BUILT_SOURCES) config.h
+-	cppcheck --enable=all --template=gcc -include config.h -I . \
++	cppcheck --enable=all --template=gcc -include src/include/config.h -I src/include \
+ 		-I src/crypto -I src/frontend -I src/network -I src/protobufs \
+ 		-I src/statesync -I src/terminal -I src/util \
+ 		-I /usr/include -I /usr/include/google/protobuf -I/usr/include/openssl \
+diff --git a/configure.ac b/configure.ac
+index 3ad983d..4518cc6 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -7,7 +7,7 @@ AM_INIT_AUTOMAKE([foreign std-options -Wall -Werror])
+ m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
+ AC_CONFIG_SRCDIR([src/frontend/mosh-client.cc])
+ AC_CONFIG_MACRO_DIR([m4])
+-AC_CONFIG_HEADERS([config.h])
++AC_CONFIG_HEADERS([src/include/config.h])
+
+ # Checks for programs.
+ AC_PROG_CC([cc gcc clang])
+@@ -491,6 +491,7 @@ AC_CONFIG_FILES([
+   src/crypto/Makefile
+   src/frontend/Makefile
+   src/network/Makefile
++  src/include/Makefile
+   src/protobufs/Makefile
+   src/statesync/Makefile
+   src/terminal/Makefile
+diff --git a/src/Makefile.am b/src/Makefile.am
+index 2390f7c..a714997 100644
+--- a/src/Makefile.am
++++ b/src/Makefile.am
+@@ -1 +1 @@
+-SUBDIRS = protobufs util crypto terminal network statesync frontend examples tests
++SUBDIRS = include protobufs util crypto terminal network statesync frontend examples tests
+diff --git a/src/include/Makefile.am b/src/include/Makefile.am
+new file mode 100644
+index 0000000..bd15b83
+--- /dev/null
++++ b/src/include/Makefile.am
+@@ -0,0 +1,14 @@
++BUILT_SOURCES = version.h
++nodist_INCLUDES = version.h
++
++clean-local:
++	@rm -rf version.h
++
++version.h:	../../VERSION
++	@test -s $<
++	@printf '#define BUILD_VERSION "%s"\n' "$$(cat $<)" > $@.new
++	@set -e; if ! diff -q $@ $@.new > /dev/null 2>&1; then \
++		mv -f $@.new $@; \
++	fi
++	@rm -f $@.new
++
```
</details>

<details>
<summary>mosh.rb diff 2 based previously mentioned renaming</summary>

```patch
diff --git a/Formula/mosh.rb b/Formula/mosh.rb
index 515ab1925a..707adf78be 100644
--- a/Formula/mosh.rb
+++ b/Formula/mosh.rb
@@ -13,6 +13,9 @@ class Mosh < Formula
       url "https://github.com/mobile-shell/mosh/commit/e5f8a826ef9ff5da4cfce3bb8151f9526ec19db0.patch?full_index=1"
       sha256 "022bf82de1179b2ceb7dc6ae7b922961dfacd52fbccc30472c527cb7c87c96f0"
     end
+
+    # Fix big sur build.
+    patch :DATA
   end

   bottle do
@@ -54,3 +57,55 @@ class Mosh < Formula
     system bin/"mosh-client", "-c"
   end
 end
+
+__END__
+diff --git a/Makefile.in b/Makefile.in
+index 2de6cc8..f77ae3b 100644
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -811,23 +811,23 @@ VERSION:
+ 	@echo @PACKAGE_STRING@ > VERSION.dist
+ 	@set -e; if git describe --dirty --always > VERSION.git 2>&1 && \
+ 		[ -z `git rev-parse --show-prefix` ]; then \
+-		if ! diff -q VERSION.git VERSION > /dev/null 2>&1; then \
+-			mv -f VERSION.git VERSION; \
++		if ! diff -q VERSION.git VERSION.txt > /dev/null 2>&1; then \
++			mv -f VERSION.git VERSION.txt; \
+ 		fi; \
+-	elif ! diff -q VERSION.dist VERSION > /dev/null 2>&1; then \
+-		mv -f VERSION.dist VERSION; \
++	elif ! diff -q VERSION.dist VERSION.txt > /dev/null 2>&1; then \
++		mv -f VERSION.dist VERSION.txt; \
+ 	fi
+ 	@rm -f VERSION.dist VERSION.git
+
+ version.h:	VERSION
+-	@printf '#define BUILD_VERSION "%s"\n' "$$(cat VERSION)" > version.h.new
++	@printf '#define BUILD_VERSION "%s"\n' "$$(cat VERSION.txt)" > version.h.new
+ 	@set -e; if ! diff -q version.h version.h.new > /dev/null 2>&1; then \
+ 		mv -f version.h.new version.h; \
+ 	fi
+ 	@rm -f version.h.new
+
+ clean-local:
+-	@rm -rf version.h VERSION cov-int mosh-coverity.txz
++	@rm -rf version.h VERSION.txt cov-int mosh-coverity.txz
+
+ cppcheck: $(BUILT_SOURCES) config.h
+ 	cppcheck --enable=all --template=gcc -include config.h -I . \
+diff --git a/scripts/Makefile.in b/scripts/Makefile.in
+index 13d261e..45261e5 100644
+--- a/scripts/Makefile.in
++++ b/scripts/Makefile.in
+@@ -513,8 +513,8 @@ uninstall-am: uninstall-binSCRIPTS
+ .PRECIOUS: Makefile
+
+
+-mosh:	mosh.pl ../VERSION Makefile
+-	@sed -e "s/\@VERSION\@/`cat ../VERSION`/" -e "s/\@PACKAGE_STRING\@/@PACKAGE_STRING@/" $(srcdir)/mosh.pl > mosh
++mosh:	mosh.pl ../VERSION.txt Makefile
++	@sed -e "s/\@VERSION\@/`cat ../VERSION.txt`/" -e "s/\@PACKAGE_STRING\@/@PACKAGE_STRING@/" $(srcdir)/mosh.pl > mosh
+ 	@chmod a+x mosh
+
+ # Tell versions [3.59,3.63) of GNU make to not export all variables.
```
</details>

**Comment by carlocab** [comment]
I think the issue is that there are multiple `generate_from_protos.sh` scripts we need to run. Some of them might live in the submodules.

**Comment by cho-m** [comment]
Looks more like a protobuf bug. For some reason, it doesn't allow `==`:
```h
#include <google/protobuf/port_def.inc>
#if PROTOBUF_VERSION < 3017000
#error This file was generated by a newer version of protoc which is
#error incompatible with your Protocol Buffer headers. Please update
#error your headers.
#endif
#if 3017000 < PROTOBUF_MIN_PROTOC_VERSION
#error This file was generated by an older version of protoc which is
#error incompatible with your Protocol Buffer headers. Please
#error regenerate this file with a newer version of protoc.
#endif
```

**Comment by cho-m** [comment]
@carlocab the `git restore` is reverting the generated files.

**Comment by carlocab** [comment]
Oops, of course it was. Silly of me to put that there. Ok.

**Comment by carlocab** [comment]
Just noticed MacPorts already have a patch for `mosh`: https://github.com/macports/macports-ports/blob/master/net/mosh/files/patch-version-subdir.diff

We may want to use that, as it looks like their patch has been around for a while (and therefore more thoroughly tested). We'd need to regenerate the `configure` script though. OTOH, it's hard to imagine the harm of getting this patch wrong...

**Comment by cho-m** [comment]
The MacPorts patch should work out.
It looks similar to what I had ended up with when I ported the upstream commit, so it aligns with what upstream decided.

Just need to use the same build steps as `head`.

**Review by carlocab — APPROVED** [review]

**Comment by Loyalsoldier** [comment]
Is the bot lost? 😂

**Comment by carlocab** [comment]
No, the PR just has `automerge-skip`. Need to be triggered manually.

**Comment by carlocab** [comment]
Triggered https://github.com/Homebrew/homebrew-core/actions/runs/853292296, but it seems to be taking a while.

**Comment by carlocab** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/53b4611aa9b83fc4f07d228e3f615858978943f3.

**Pull request closed without merging**